src/evolve_energy.cxx

```diff
@@ -288,31 +288,23 @@ void EvolveEnergy::finally(const Options& state) {
 
           std::string collision_name = collision.second.name();
 
-          if (identifySpeciesType(species.name()) == "neutral") {
-            if (/// Charge exchange
-                (collisionSpeciesMatch(    
-                  collision_name, species.name(), "+", "cx", "partial")) or
-                /// Ionisation
-                (collisionSpeciesMatch(    
-                  collision_name, species.name(), "+", "iz", "partial"))) {
-                    collision_names.push_back(collision_name);
-                  }
-
-          } else if (identifySpeciesType(species.name()) == "electron") {
+          if (identifySpeciesTypeEnum(species.name()) == SpeciesType::neutral) {
```

Comment:
`species.name()` gets called a lot, as does `identifySpeciesTypeEnum`, so we should add `const` local variables.

Maybe it would even be helpful to have a `Species` class? (That's obviously bigger than this PR)

Also, now that this `if` statement is over an enum, we can use a `switch` statement instead:

```cpp
const auto species_name = species.name();
const auto species_type = identifySpeciesTypeEnum(species_name);

switch (species_type) {
case SpeciesType::neutral:
  // ...
case SpeciesType::electron:
  // ...
case SpeciesType::ion:
  // ...
};
```

This means the compiler can catch unhandled values (with `-Wswitch-enum`, we could include this by default)